pkg/webhook/work/mutating.go

```diff
@@ -80,3 +83,21 @@ func (a *MutatingAdmission) Handle(_ context.Context, req admission.Request) adm
 
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledBytes)
 }
+
+// setAssociatedLabelsAndAnnotations sets the associated work object labels and annotations.
+func setAssociatedLabelsAndAnnotations(workload *unstructured.Unstructured, work *workv1alpha1.Work) {
+	workload.SetAnnotations(util.DedupeAndMergeAnnotations(workload.GetAnnotations(), map[string]string{
+		workv1alpha2.ResourceTemplateUIDAnnotation: string(workload.GetUID()),
+		workv1alpha2.WorkNamespaceAnnotation:       work.GetNamespace(),
+		workv1alpha2.WorkNameAnnotation:            work.GetName(),
+	}))
+	workload.SetLabels(util.DedupeAndMergeLabels(workload.GetLabels(), map[string]string{
+		util.ManagedByKarmadaLabel:        util.ManagedByKarmadaLabelValue,
+		workv1alpha2.WorkPermanentIDLabel: work.Labels[workv1alpha2.WorkPermanentIDLabel],
+		// TODO(whitewindmills): Delete the following two lines in a future version.
+		workv1alpha2.WorkNamespaceLabel: work.GetNamespace(),
+		workv1alpha2.WorkNameLabel:      work.GetName(),
```

Comment:
@XiShanYongYe-Chang 
These labels and annotations are spread out in many places, which is very messy. Let's focus on these labels and annotations here, I think it will be easier to maintain.

Reply:
Sorry, I didn't get this. The annotations for namespace/name should be maintained later, I don't get why they were removed.